### ISSUE TITLE
incorrect identification of leap years; multiple implementations. 

### ISSUE THREAD

**Issue by dlebauer** [PecanProject/pecan#801]
The logic to test if a year is a leap year has been implemented perhaps a dozen times independently [within the PEcAn code base](https://github.com/PecanProject/pecan/search?utf8=%E2%9C%93&q=leap), including once as the function [`is.leap`](https://github.com/PecanProject/pecan/blob/1743b942fdba4adb7cd6cc7716f2f3ae258eb2c0/models/ed/inst/timeutils.R).
2. there are [many cases where `year %% 4 i== 0`](https://github.com/PecanProject/pecan/search?utf8=%E2%9C%93&q=%22year+%25%25+4%22&type=Code) is used although it should be `year %% 400 == 0 | (year %% 4 == 0 & year %% 100 != 0)` (`is.leap` and `leap_year` implement this correctly).
1. The lubridate package has a function `leap_year` that returns T/F. We already use this function [in many places](https://github.com/PecanProject/pecan/search?utf8=%E2%9C%93&q=%22leap_year%22&type=Code) and I suggest that this is a good solution
3. (just a comment / food for thought) testing if a year is a leap year is often followed by calculating time steps (e.g. days or seconds/year). `lubridate` also provides convenience functions for this that would be less error prone than declaring seconds / year or days / month constants repeatedly in the code.

TODO
- [ ] _bugfix:_ All of the invalid uses of `year %% 4 i== 0` should be replaced by a valid function (`leap_year`)
- [ ] _enhancement:_ deprecate `is.leap` and replace it with `lubridate::leap_year`
- [ ] _documentation_ add some notes about existing 'helper' functions to the model template README

**Comment by robkooper** [comment]
Other handy functions to add:

```
days_in_year <- function(year) {
  ifelse(lubridate:: leap_year(year), 366, 365)
}

hours_in_year <- function(year, every=24) {
  24/every * ifelse(lubridate:: leap_year(year), 366, 365)
}
```

now we can simply call `24/6 * days_in_year(1999)`` to get the value for every 6 hours.

**Comment by ashiklom** [comment]
Closed by #1623.
